File: CRO/pyCRO/RO_BWJ.py
import numpy as np

def RO_BWJ(par):
    # Extract annual mean values
    R_value = par['R'][0]
    F1_value = par['F1'][0]
    epsilon_value = par['epsilon'][0]
    F2_value = par['F2'][0]

    # Calculate growth rate and frequency
    gr = (R_value - epsilon_value) / 2
    w = np.sqrt(4 * F1_value * F2_value - (R_value + epsilon_value)**2) / 2

    # Return complex index
    return gr + 1j * w

File: CRO/mCRO/RO_BWJ.m
function BWJ=RO_BWJ(par)
%% Calculate BWJ index from Par 
%% Setting Parameter
% post-processing par
for i=1:length(par)
    par_sel(i)=par{i}(1); % only select annual mean value
end 

% linear parameter
R=par_sel(1);
F1=par_sel(2);
F2=par_sel(3);
epsilon=par_sel(4);

%% Calculate BWJ
gr=(R-epsilon)/2;
w=sqrt(4*F1*F2-(R+epsilon)^2)/2;
BWJ=gr+j*w;

end 